sys/psa_crypto/psa_hmac.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2024 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_psa_crypto
+ * @defgroup    sys_psa_crypto_hmac  PSA Generic HMAC
+ * @{
+ *
+ * @file        psa_hmac.c
+ * @brief       Generic PSA Crypto HMAC implementation.
```

Comment:
Maybe add a link to the RFC this is implemented after.

Reply:
I am not sure putting this implementation into a `psa_hmac.c` file is the way to go.
IIRC we were planning on splitting `psa_crypto.c` up into separate modules. Then we'd end up with a `psa_mac.c` and a `psa_hmac.c`, one containing the user facing API and one containing this lower level API.

Maybe it would be better to have an additional source folder for generic/builtin implementations, or, if we keep them here, at least call them that (e.g. `psa_generic_hmac.c`)?

Also I would then call the functions `psa_generic_hmac_setup_key` etc. 
Otherwise, to me, it's unclear that these are not part of the user facing API :)

Reply:
Ok